### PR TITLE
add documentation for Azure Monitor Pod Identity

### PR DESCRIPTION
**Pull request by privalof**
Closes https://github.com/kedacore/keda/issues/936

### CODE REVIEW

File: content/docs/2.0/scalers/azure-monitor.md [hunk 1]
```diff
@@ -43,7 +43,8 @@ triggers:
 
 ### Authentication Parameters
 
-You can use `TriggerAuthentication` CRD to configure the authentication by providing a set of Azure Active Directory credentials.
+You can use `TriggerAuthentication` CRD to configure the authentication by providing a set of Azure Active Directory credentials
+or by using pod identity.
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/azure-monitor.md:47]: Please move this on the line above

File: content/docs/2.0/scalers/azure-monitor.md [hunk 2]
```diff
@@ -75,6 +76,9 @@ spec:
     - parameter: activeDirectoryClientPassword
       name: azure-monitor-secrets
       key: activeDirectoryClientPassword
+  # or Pod Identity, kind: Secret is not required in case of pod Identity
+  podIdentity:
+      provider: azure
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/azure-monitor.md:81]:
```suggestion
      provider: azure
```

File: content/docs/2.0/scalers/azure-monitor.md [hunk 3]
```diff
@@ -23,8 +23,8 @@ triggers:
     metricFilter: namespace eq 'default'
     metricAggregationInterval: "0:1:0"
     targetValue: "1"
-    activeDirectoryClientId: CLIENT_ID_ENV_NAME
-    activeDirectoryClientPassword: CLIENT_PASSWORD_ENV_NAME
+    activeDirectoryClientId: CLIENT_ID_ENV_NAME # Optional if podIdentity is defined
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/azure-monitor.md:26]: Let's be more precise:
```suggestion
    activeDirectoryClientId: CLIENT_ID_ENV_NAME # Optional, can use TriggerAuthentication as well
```

File: content/docs/2.0/scalers/azure-monitor.md [hunk 4]
```diff
@@ -23,8 +23,8 @@ triggers:
...
+    activeDirectoryClientPassword: CLIENT_PASSWORD_ENV_NAME # Optional if podIdentity is defined
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/azure-monitor.md:27]: Let's be more precise:

```suggestion
    activeDirectoryClientPassword: CLIENT_PASSWORD_ENV_NAME # Optional, can use TriggerAuthentication as well
```

**Comment by tomkerkhove** [comment]
Thanks!

File: content/docs/2.0/scalers/azure-monitor.md (hunk 1 above)
Comment (privalof) [reply content/docs/2.0/scalers/azure-monitor.md]: done

**Pull request merged**